### PR TITLE
Return caf::message from command::run

### PR DESCRIPTION
**Pull request by Neverlord**
This allows returning errors as well as actual results from commands.

### CODE REVIEW

**Comment by Neverlord** [comment]
Btw, this also fixes the bug that subcommands don't have help texts.

File: libvast/vast/command.hpp [hunk 1]
```diff
@@ -22,11 +22,11 @@
 #include <caf/actor_system_config.hpp>
 #include <caf/config_option_set.hpp>
 #include <caf/fwd.hpp>
-#include <caf/message.hpp>
 #include <caf/pec.hpp>
 
 #include "vast/data.hpp"
 #include "vast/error.hpp"
+#include "vast/error.hpp"
```
Comment (tobim) [inline libvast/vast/command.hpp]: ?
Also, why remove caf/message?
And while we're at it: `vast/data.hpp`, `vast/detail/steady_map.hpp` can be removed. `vast/detail/string.hpp` can be moved to the implementation. `caf::config_value_map` can be forward declared (or added to `caf/fwd.hpp`). `caf/actor_system_config.hpp` can be removed also, and `functional` is unused as well.
There will be some more fallout, but that can be easily fixed.

Comment (Neverlord) [reply libvast/vast/command.hpp]: Headers cleaned up.

> Also, why remove caf/message?

We only need a forward declaration to `caf::message` in `command.hpp`.

Comment (tobim) [reply libvast/vast/command.hpp]: I see. I think forward declared return types are a quite subtle issue.
It avoids the cost of `message.hpp` for TUs that don't need the definition. But puts the burden of providing the definition at TUs that do call `run()`.
Two files to so: `application.cpp` and `default_application.cpp`
- `application` does call `run()`, but misses the include for `message` (its definition is provided by an unrelated include chain).
- `default_application` does not call `run()`. *So this justifies it*

After this I have one more request: Add a direct include of `caf/message.hpp` to `application.cpp`.

File: libvast/vast/system/application.hpp [hunk 2]
```diff
@@ -48,8 +49,8 @@ class application {
 
   /// Starts the application and blocks until execution completes.
   /// @returns An exit code suitable for returning from main.
```
Comment (tobim) [inline libvast/vast/system/application.hpp]: This comment needs an update too.

File: libvast/vast/command.hpp [hunk 3]
```diff
@@ -59,14 +45,13 @@ class command {
 
   /// Runs the command and blocks until execution completes.
   /// @returns An exit code suitable for returning from main.
```
Comment (tobim) [inline libvast/vast/command.hpp]: This comment needs an update too.

File: libvast/vast/command.hpp [hunk 4]
```diff
@@ -59,14 +45,13 @@ class command {
...
-  int run(caf::actor_system& sys, argument_iterator begin,
-          argument_iterator end);
+  caf::message run(caf::actor_system& sys, argument_iterator begin,
+                   argument_iterator end);
 
   /// Runs the command and blocks until execution completes.
   /// @returns An exit code suitable for returning from main.
```
Comment (tobim) [inline libvast/vast/command.hpp]: This comment needs an update too.

File: libvast/vast/command.hpp (hunk 1 above)
Comment (Neverlord) [reply libvast/vast/command.hpp]: `application.cpp` already includes `caf/all.hpp`. There's no need to include any other CAF header for core functionality. We could argue whether it should pull in everything, but that's orthogonal to this PR IMO.

Comment (tobim) [reply libvast/vast/command.hpp]: ah ok, sorry I missed that.

**Pull request merged**